Loading of dynamic extensions when using --libc=glibc

.env.ini override:
```ini
SPC_DEFAULT_C_FLAGS=-fPIC
SPC_NO_MUSL_PATH=yes
CC=gcc
CXX=g++
AR=ar
LD=ld.gold
SPC_DEFAULT_C_FLAGS=-fPIE
EXTENSION_DIR=/usr/lib64/php/modules
SPC_CMD_PREFIX_PHP_CONFIGURE="${SPC_PHP_DEFAULT_LD_LIBRARY_PATH_CMD} ./configure --prefix= --with-valgrind=no --enable-shared=yes --enable-static=yes --disable-all --disable-cgi --disable-phpdbg --with-pic"
SPC_CMD_VAR_PHP_MAKE_EXTRA_LIBS="-ldl -lpthread -lm"
; EXTRA_LDFLAGS_PROGRAM for `make` php
SPC_CMD_VAR_PHP_MAKE_EXTRA_LDFLAGS_PROGRAM="-Wl,-O1 -pie"
```

Building with:
```console
bin/spc build --build-cli --build-fpm --build-embed "apcu,bcmath,calendar,ctype,curl,dba,dom,exif,fileinfo,filter,gd,iconv,intl,ldap,mbregex,mbstring,mysqli,mysqlnd,opcache,openssl,pcntl,pdo,pdo_mysql,pdo_sqlite,phar,posix,readline,redis,session,simplexml,sockets,sodium,sqlite3,tokenizer,xml,xmlreader,xmlwriter,xsl,zip,zlib" --enable-zts --with-config-file-path="/etc" --with-config-file-scan-dir="/etc/php.d"
```

Unfortunate, the result of attempting to load dynamic extensions:
```console
[root@svmdev static-php-cli]# buildroot/bin/php -v
PHP Warning:  Failed loading Zend extension 'opcache' (tried: /usr/lib64/php/modules/opcache (/usr/lib64/php/modules/opcache: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/opcache.so (/usr/lib64/php/modules/opcache.so: undefined symbol: compiler_globals)) in Unknown on line 0
PHP Warning:  Failed loading Zend extension 'xdebug-zts.so' (tried: /usr/lib64/php/modules/xdebug-zts.so (/usr/lib64/php/modules/xdebug-zts.so: undefined symbol: compiler_globals), /usr/lib64/php/modules/xdebug-zts.so.so (/usr/lib64/php/modules/xdebug-zts.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
PHP Warning:  PHP Startup: bz2: Unable to initialize module
Module compiled with build ID=API20240924,NTS
PHP    compiled with build ID=API20240924,TS
These options need to match
 in Unknown on line 0
PHP Warning:  Module "calendar" is already loaded in Unknown on line 0
PHP Warning:  Module "ctype" is already loaded in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'curl' (tried: /usr/lib64/php/modules/curl (/usr/lib64/php/modules/curl: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/curl.so (/usr/lib64/php/modules/curl.so: undefined symbol: core_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'dom' (tried: /usr/lib64/php/modules/dom (/usr/lib64/php/modules/dom: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/dom.so (/usr/lib64/php/modules/dom.so: undefined symbol: executor_globals)) in Unknown on line 0
PHP Warning:  Module "exif" is already loaded in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'fileinfo' (tried: /usr/lib64/php/modules/fileinfo (/usr/lib64/php/modules/fileinfo: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/fileinfo.so (/usr/lib64/php/modules/fileinfo.so: undefined symbol: file_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'ftp' (tried: /usr/lib64/php/modules/ftp (/usr/lib64/php/modules/ftp: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/ftp.so (/usr/lib64/php/modules/ftp.so: undefined symbol: compiler_globals)) in Unknown on line 0
PHP Warning:  Module "gd" is already loaded in Unknown on line 0
PHP Warning:  PHP Startup: gettext: Unable to initialize module
Module compiled with build ID=API20240924,NTS
PHP    compiled with build ID=API20240924,TS
These options need to match
 in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'iconv' (tried: /usr/lib64/php/modules/iconv (/usr/lib64/php/modules/iconv: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/iconv.so (/usr/lib64/php/modules/iconv.so: undefined symbol: sapi_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'intl' (tried: /usr/lib64/php/modules/intl (/usr/lib64/php/modules/intl: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/intl.so (/usr/lib64/php/modules/intl.so: undefined symbol: executor_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'ldap' (tried: /usr/lib64/php/modules/ldap (/usr/lib64/php/modules/ldap: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/ldap.so (/usr/lib64/php/modules/ldap.so: undefined symbol: compiler_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'mbstring' (tried: /usr/lib64/php/modules/mbstring (/usr/lib64/php/modules/mbstring: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/mbstring.so (/usr/lib64/php/modules/mbstring.so: undefined symbol: sapi_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'mysqlnd' (tried: /usr/lib64/php/modules/mysqlnd (/usr/lib64/php/modules/mysqlnd: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/mysqlnd.so (/usr/lib64/php/modules/mysqlnd.so: undefined symbol: core_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'pdo' (tried: /usr/lib64/php/modules/pdo (/usr/lib64/php/modules/pdo: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/pdo.so (/usr/lib64/php/modules/pdo.so: undefined symbol: executor_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'phar' (tried: /usr/lib64/php/modules/phar (/usr/lib64/php/modules/phar: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/phar.so (/usr/lib64/php/modules/phar.so: undefined symbol: compiler_globals)) in Unknown on line 0
PHP Warning:  Module "posix" is already loaded in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'shmop' (tried: /usr/lib64/php/modules/shmop (/usr/lib64/php/modules/shmop: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/shmop.so (/usr/lib64/php/modules/shmop.so: undefined symbol: compiler_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'simplexml' (tried: /usr/lib64/php/modules/simplexml (/usr/lib64/php/modules/simplexml: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/simplexml.so (/usr/lib64/php/modules/simplexml.so: undefined symbol: executor_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'sodium' (tried: /usr/lib64/php/modules/sodium (/usr/lib64/php/modules/sodium: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/sodium.so (/usr/lib64/php/modules/sodium.so: undefined symbol: compiler_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'sqlite3' (tried: /usr/lib64/php/modules/sqlite3 (/usr/lib64/php/modules/sqlite3: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/sqlite3.so (/usr/lib64/php/modules/sqlite3.so: undefined symbol: core_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'sysvmsg' (tried: /usr/lib64/php/modules/sysvmsg (/usr/lib64/php/modules/sysvmsg: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/sysvmsg.so (/usr/lib64/php/modules/sysvmsg.so: undefined symbol: executor_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: sysvsem: Unable to initialize module
Module compiled with build ID=API20240924,NTS
PHP    compiled with build ID=API20240924,TS
These options need to match
 in Unknown on line 0
PHP Warning:  PHP Startup: sysvshm: Unable to initialize module
Module compiled with build ID=API20240924,NTS
PHP    compiled with build ID=API20240924,TS
These options need to match
 in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'tokenizer' (tried: /usr/lib64/php/modules/tokenizer (/usr/lib64/php/modules/tokenizer: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/tokenizer.so (/usr/lib64/php/modules/tokenizer.so: undefined symbol: compiler_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'xml' (tried: /usr/lib64/php/modules/xml (/usr/lib64/php/modules/xml: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/xml.so (/usr/lib64/php/modules/xml.so: undefined symbol: executor_globals)) in Unknown on line 0
PHP Warning:  Module "xmlwriter" is already loaded in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'xsl' (tried: /usr/lib64/php/modules/xsl (/usr/lib64/php/modules/xsl: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/xsl.so (/usr/lib64/php/modules/xsl.so: undefined symbol: executor_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'mysqli' (tried: /usr/lib64/php/modules/mysqli (/usr/lib64/php/modules/mysqli: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/mysqli.so (/usr/lib64/php/modules/mysqli.so: undefined symbol: compiler_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'pdo_mysql' (tried: /usr/lib64/php/modules/pdo_mysql (/usr/lib64/php/modules/pdo_mysql: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/pdo_mysql.so (/usr/lib64/php/modules/pdo_mysql.so: undefined symbol: executor_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'pdo_sqlite' (tried: /usr/lib64/php/modules/pdo_sqlite (/usr/lib64/php/modules/pdo_sqlite: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/pdo_sqlite.so (/usr/lib64/php/modules/pdo_sqlite.so: undefined symbol: core_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'xmlreader' (tried: /usr/lib64/php/modules/xmlreader (/usr/lib64/php/modules/xmlreader: cannot open shared object file: No such file or directory), /usr/lib64/php/modules/xmlreader.so (/usr/lib64/php/modules/xmlreader.so: undefined symbol: executor_globals)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'zip.so' (tried: /usr/lib64/php/modules/zip.so (/usr/lib64/php/modules/zip.so: undefined symbol: compiler_globals), /usr/lib64/php/modules/zip.so.so (/usr/lib64/php/modules/zip.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
PHP Warning:  PHP Startup: Unable to load dynamic library 'apcu.so' (tried: /usr/lib64/php/modules/apcu.so (/usr/lib64/php/modules/apcu.so: undefined symbol: zend_signal_globals), /usr/lib64/php/modules/apcu.so.so (/usr/lib64/php/modules/apcu.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
PHP 8.4.4 (cli) (built: Mar 11 2025 13:01:51) (ZTS)
Copyright (c) The PHP Group
Zend Engine v4.4.4, Copyright (c) Zend Technologies
    with Zend OPcache v8.4.4, Copyright (c), by Zend Technologies
```

Most of these failures are expected, as they're already built into the php executable,

But this one is not:
```console
PHP Warning:  Failed loading Zend extension 'xdebug-zts.so' (tried: /usr/lib64/php/modules/xdebug-zts.so (/usr/lib64/php/modules/xdebug-zts.so: undefined symbol: compiler_globals), /usr/lib64/php/modules/xdebug-zts.so.so (/usr/lib64/php/modules/xdebug-zts.so.so: cannot open shared object file: No such file or directory)) in Unknown on line 0
PHP Warning:  PHP Startup: bz2: Unable to initialize module
Module compiled with build ID=API20240924,NTS
PHP    compiled with build ID=API20240924,TS
These options need to match
 in Unknown on line 0
```

xdebug-zts.so was just built by me using pkg config pointing to /opt/static-php-cli/buildroot.

Edit:

Okay, I think the issue itself is almost resolved, but we should fix out php-config output prefix and properly document it.

buildroot/bin/php-config had the prefix empty, which resulted in xdebug ./configure looking at the NTS headers from the system wide php installation, effectively building xdebug for NTS php again.

Patching php-config to point to BUILD_ROOT_PATH:

```console
[root@svmdev static-php-cli]# buildroot/bin/php -v
... blablabla most libs can't be loaded ...
PHP 8.4.4 (cli) (built: Mar 11 2025 13:01:51) (ZTS)
Copyright (c) The PHP Group
Zend Engine v4.4.4, Copyright (c) Zend Technologies
    with Xdebug v3.5.0-dev, Copyright (c) 2002-2024, by Derick Rethans
    with Zend OPcache v8.4.4, Copyright (c), by Zend Technologies
```

```console
 ___| |_ __ _| |_(_) ___      _ __ | |__  _ __
/ __| __/ _` | __| |/ __|____| '_ \| '_ \| '_ \
\__ \ || (_| | |_| | (_|_____| |_) | | | | |_) |
|___/\__\__,_|\__|_|\___|    | .__/|_| |_| .__/   v2.5.0
                             |_|         |_|
[04:26:27] [I] argv: bin/spc build intl,ffi --enable-zts --build-embed --build-cli --debug
[04:26:27] [I] Setting env: SPC_LIBC=musl-wrapper
[04:26:27] [I] Setting env: PATH=/app/buildroot/bin:/app/bin:/cmake/bin:/root/.local/bin:/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
[04:26:27] [I] Setting env: PKG_CONFIG=/app/buildroot/bin/pkg-config
[04:26:27] [I] Setting env: PKG_CONFIG_PATH=/app/buildroot/lib/pkgconfig
[04:26:27] [I] Setting env: SPC_PHP_DEFAULT_OPTIMIZE_CFLAGS=-g -fstack-protector-strong -fpic -fpie -Os -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
[04:26:27] [I] Setting env: SPC_LINUX_DEFAULT_CC=x86_64-linux-musl-gcc
[04:26:27] [I] Setting env: SPC_LINUX_DEFAULT_CXX=x86_64-linux-musl-g++
[04:26:27] [I] Setting env: SPC_LINUX_DEFAULT_AR=x86_64-linux-musl-ar
[04:26:27] [I] Setting env: SPC_CONCURRENCY=6
[04:26:27] [I] Setting env: SPC_SKIP_PHP_VERSION_CHECK=no
[04:26:27] [I] Setting env: EXTENSION_DIR=
[04:26:27] [I] Setting env: SPC_DEFAULT_CXX_FLAGS=
[04:26:27] [I] Setting env: SPC_EXTRA_LIBS=
[04:26:27] [I] Setting env: UPX_EXEC=/app/pkgroot/bin/upx
[04:26:27] [I] Setting env: SPC_MICRO_PATCHES=static_extensions_win32,cli_checks,disable_huge_page,vcruntime140,win32,zend_stream
[04:26:27] [I] Setting env: SPC_CMD_PREFIX_PHP_BUILDCONF=./buildconf --force
[04:26:27] [I] Setting env: SPC_CMD_PREFIX_PHP_CONFIGURE=./configure --prefix= --with-valgrind=no --enable-shared=no --enable-static=yes --disable-all --disable-cgi --disable-phpdbg --with-pic
[04:26:27] [I] Setting env: SPC_CMD_PREFIX_PHP_MAKE=make -j6
[04:26:27] [I] Setting env: SPC_CMD_VAR_PHP_EMBED_TYPE=static
[04:26:27] [I] Setting env: SPC_CMD_VAR_PHP_CONFIGURE_CFLAGS=-fPIE -fPIC -fPIE
[04:26:27] [I] Setting env: SPC_CMD_VAR_PHP_CONFIGURE_CPPFLAGS=-I/app/buildroot/include
[04:26:27] [I] Setting env: SPC_CMD_VAR_PHP_CONFIGURE_LDFLAGS=-L/app/buildroot/lib
[04:26:27] [I] Setting env: SPC_CMD_VAR_PHP_CONFIGURE_LIBS=-ldl -lpthread -lm
[04:26:27] [I] Setting env: SPC_CMD_VAR_PHP_MAKE_EXTRA_CFLAGS=-g -fstack-protector-strong -fpic -fpie -Os -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -fno-ident -fPIE -fPIC
[04:26:27] [I] making cmake tool chain file for Linux x86_64 with CFLAGS='-fPIE -fPIC'
[04:26:27] [I] Dir /app/buildroot/lib/pkgconfig already exists, ignored
[04:26:27] [I] Dir /app/buildroot/include already exists, ignored
[04:26:27] [I] Init var [BUILD_ROOT_PATH]:                    /app/buildroot
[04:26:27] [I] Init var [BUILD_INCLUDE_PATH]:                 /app/buildroot/include
[04:26:27] [I] Init var [BUILD_LIB_PATH]:                     /app/buildroot/lib
[04:26:27] [I] Init var [BUILD_BIN_PATH]:                     /app/buildroot/bin
[04:26:27] [I] Init var [PKG_ROOT_PATH]:                      /app/pkgroot
[04:26:27] [I] Init var [SOURCE_PATH]:                        /app/source
[04:26:27] [I] Init var [DOWNLOAD_PATH]:                      /app/downloads
[04:26:27] [I] Init var [CPU_COUNT]:                          6
[04:26:27] [I] Init var [GNU_ARCH]:                           x86_64
[04:26:27] [I] Init var [MAC_ARCH]:                           x86_64
[04:26:27] [I] Init var [PATH]:                               /app/buildroot/bin:/app/bin:/cmake/bin:/root/.local/bin:/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
[04:26:27] [I] Init var [PKG_CONFIG]:                         /app/buildroot/bin/pkg-config
[04:26:27] [I] Init var [PKG_CONFIG_PATH]:                    /app/buildroot/lib/pkgconfig
[04:26:27] [I] Init var [SPC_PHP_DEFAULT_OPTIMIZE_CFLAGS]:    -g -fstack-protector-strong -fpic -fpie -Os -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
[04:26:27] [I] Init var [SPC_LINUX_DEFAULT_CC]:               x86_64-linux-musl-gcc
[04:26:27] [I] Init var [SPC_LINUX_DEFAULT_CXX]:              x86_64-linux-musl-g++
[04:26:27] [I] Init var [SPC_LINUX_DEFAULT_AR]:               x86_64-linux-musl-ar
[04:26:27] [I] Init var [SPC_CONCURRENCY]:                    6
[04:26:27] [I] Init var [SPC_SKIP_PHP_VERSION_CHECK]:         no
[04:26:27] [I] Init var [EXTENSION_DIR]:
[04:26:27] [I] Init var [SPC_DEFAULT_CXX_FLAGS]:
[04:26:27] [I] Init var [SPC_EXTRA_LIBS]:
[04:26:27] [I] Init var [UPX_EXEC]:                           /app/pkgroot/bin/upx
[04:26:27] [I] Init var [SPC_MICRO_PATCHES]:                  static_extensions_win32,cli_checks,disable_huge_page,vcruntime140,win32,zend_stream
[04:26:27] [I] Init var [SPC_CMD_PREFIX_PHP_BUILDCONF]:       ./buildconf --force
[04:26:27] [I] Init var [SPC_CMD_PREFIX_PHP_CONFIGURE]:       ./configure --prefix= --with-valgrind=no --enable-shared=no --enable-static=yes --disable-all --disable-cgi --disable-phpdbg --with-pic
[04:26:27] [I] Init var [SPC_CMD_PREFIX_PHP_MAKE]:            make -j6
[04:26:27] [I] Init var [SPC_CMD_VAR_PHP_EMBED_TYPE]:         static
[04:26:27] [I] Init var [SPC_CMD_VAR_PHP_CONFIGURE_CFLAGS]:   -fPIE -fPIC -fPIE
[04:26:27] [I] Init var [SPC_CMD_VAR_PHP_CONFIGURE_CPPFLAGS]: -I/app/buildroot/include
[04:26:27] [I] Init var [SPC_CMD_VAR_PHP_CONFIGURE_LDFLAGS]:  -L/app/buildroot/lib
[04:26:27] [I] Init var [SPC_CMD_VAR_PHP_CONFIGURE_LIBS]:     -ldl -lpthread -lm
[04:26:27] [I] Init var [SPC_CMD_VAR_PHP_MAKE_EXTRA_CFLAGS]:  -g -fstack-protector-strong -fpic -fpie -Os -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -fno-ident -fPIE -fPIC
[04:26:27] [I] Build OS:         Linux (x86_64)
[04:26:27] [I] Build SAPI:       cli, embed
[04:26:27] [I] Extensions (2):   intl,ffi
[04:26:27] [I] Libraries (6):    pkg-config,icu,libffi
[04:26:27] [I] Strip Binaries:   yes
[04:26:27] [I] Enable ZTS:       yes
[04:26:27] [I] Config File Path: /usr/local/etc/php
[04:26:27] [I] PHP Version:      8.4.4
[04:26:27] [I] Build will start after 2s ...
[04:26:29] [I] scanning directory /app/src/SPC/builder/linux/library
[04:26:29] [I] scanning directory /app/src/SPC/builder/extension
[04:26:29] [I] Source [php-src] already extracted in /app/source/php-src, skip !
[04:26:29] [I] Source [micro] already extracted in /app/source/php-src/sapi/micro, skip !
[04:26:29] [I] Building required package [pkg-config]
[04:26:29] [I] Entering dir: /app/source/pkg-config
[04:26:29] [I] [EXEC] CFLAGS="-fPIE -fPIC " LDFLAGS=" --static" ./configure --disable-shared --enable-static --with-internal-glib --disable-host-tool --with-pic --prefix=/app/buildroot --without-sysroot --without-system-include-path --without-system-library-path --without-pc-path
[04:26:29] [I] Executed at: /app/src/SPC/util/UnixShell.php:88
[04:26:29] [I] [PASSTHRU] cd '/app/source/pkg-config' && CFLAGS="-fPIE -fPIC " LDFLAGS=" --static" ./configure --disable-shared --enable-static --with-internal-glib --disable-host-tool --with-pic --prefix=/app/buildroot --without-sysroot --without-system-include-path --without-system-library-path --without-pc-path
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for style of include used by make... GNU
checking for gcc... /opt/rh/devtoolset-10/root/usr/bin/gcc
checking whether the C compiler works... no
configure: error: in `/app/source/pkg-config':
configure: error: C compiler cannot create executables
See `config.log' for more details
[04:26:30] [I] Uncaught SPC\exception\RuntimeException: Command run failed with code[77]: cd '/app/source/pkg-config' && CFLAGS="-fPIE -fPIC " LDFLAGS=" --static" ./configure --disable-shared --enable-static --with-internal-glib --disable-host-tool --with-pic --prefix=/app/buildroot --without-sysroot --without-system-include-path --without-system-library-path --without-pc-path at /app/src/globals/functions.php(166)
[04:26:30] [I] #0 /app/src/SPC/util/UnixShell.php(51): f_passthru('cd '/app/source...')
#1 /app/src/SPC/util/UnixShell.php(88): SPC\util\UnixShell->exec('cd '/app/source...')
#2 /app/src/SPC/builder/unix/library/pkgconfig.php(18): SPC\util\UnixShell->execWithEnv('./configure --d...')
#3 /app/src/SPC/builder/LibraryBase.php(231): SPC\builder\linux\library\pkgconfig->build()
#4 /app/src/SPC/builder/LibraryBase.php(255): SPC\builder\LibraryBase->tryBuild(true)
#5 /app/src/SPC/builder/LibraryBase.php(51): SPC\builder\LibraryBase->tryBuild(false)
#6 /app/src/SPC/builder/BuilderBase.php(66): SPC\builder\LibraryBase->setup(false)
#7 /app/src/SPC/command/BuildCliCommand.php(162): SPC\builder\BuilderBase->setupLibs()
#8 /app/src/SPC/command/BaseCommand.php(106): SPC\command\BuildCliCommand->handle()
#9 /app/vendor/symfony/console/Command/Command.php(326): SPC\command\BaseCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /app/vendor/symfony/console/Application.php(1078): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /app/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand(Object(SPC\command\BuildCliCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /app/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /app/bin/spc(31): Symfony\Component\Console\Application->run()```

I think it's trying to use the wrong compiler? Do I need to change the config in the host system?

--- a/bin/build-static-frankenphp
+++ b/bin/build-static-frankenphp
@@ -104,6 +104,14 @@ RUN curl -o make.tgz -fsSL https://ftp.gnu.org/gnu/make/make-4.4.tar.gz && \
     make install && \
     ln -sf /usr/local/bin/make /usr/bin/make
 
+RUN curl -o automake.tgz -fsSL https://ftp.gnu.org/gnu/automake/automake-1.17.tar.xz && \
+    tar -xvf automake.tgz && \
+    cd automake-1.17 && \
+    ./configure && \
+    make && \
+    make install && \
+    ln -sf /usr/local/bin/automake /usr/bin/automake
+
 RUN git clone https://github.com/static-php/gnu-frankenphp --depth=1 /frankenphp
 WORKDIR /frankenphp
 

--- a/bin/spc-gnu-docker
+++ b/bin/spc-gnu-docker
@@ -100,6 +100,14 @@ RUN curl -o make.tgz -fsSL https://ftp.gnu.org/gnu/make/make-4.4.tar.gz && \
     make install && \
     ln -sf /usr/local/bin/make /usr/bin/make
 
+RUN curl -o automake.tgz -fsSL https://ftp.gnu.org/gnu/automake/automake-1.17.tar.xz && \
+    tar -xvf automake.tgz && \
+    cd automake-1.17 && \
+    ./configure && \
+    make && \
+    make install && \
+    ln -sf /usr/local/bin/automake /usr/bin/automake
+
 EOF
 fi
 

